### ISSUE TITLE
Using mockedDBContext in dependency injection

### ISSUE THREAD

**Issue by brandoq123** [rgvlee/EntityFrameworkCore.Testing#38]
Is there support or a way for us to use Create.MockedDbContextFor in dependency injection? 
I am trying to find a way to register the context to my ServiceCollection but never come up with anything that actually works.

Services.AddDbContext does not have any way to add the mockedDbContext and registering the context manually via Services.AddTransient(c => mockedDbContext) doesn't work as well.

Basically I am trying to do something like this for the mockedDbContext.

`
Services.AddDbContext<MainDbContext>(opt =>
                    opt.UseInMemoryDatabase(databaseName: "mainDb"),
                ServiceLifetime.Transient);
`

**Comment by rgvlee** [comment]
Does this previous issue/answer answer your question? https://github.com/rgvlee/EntityFrameworkCore.Testing/issues/37

**Comment by brandoq123** [comment]
That does not work. Whenever I try to use the db context it throws and exception saying the DbContext has not be registered.

**Comment by rgvlee** [comment]
The reason for referring to the previous issue/answer is to verify the reason for using the mocked db context in this way.
What is the use case for using a service provider in your unit tests?
Are you unit testing or integration testing? If you are unit testing, lifetime scope shouldn't be a concern.
If you are integration testing, you wouldn't use an in-memory or mocked db context.

To answer the question directly, it is possible to use the mocked db context with a service provider. See the following LINQPad working example: https://gist.github.com/rgvlee/e7683d997c29076f0be9b1fc0447f1c4

Execute result:
![image](https://user-images.githubusercontent.com/47836784/103450572-75ad7d00-4cf3-11eb-98f6-2dae6d8d09d8.png)

I don't recommend using it this way. This is using the same database name to persist state between scopes and as this is an in-memory provider behaviour, I cannot guarantee it's usage/behaviour.

If you are still having trouble, please provide a minimum reproducible example so I can assist further.

**Comment by brandoq123** [comment]
Thanks for the response. They way we would be using it is closer to an integration test because we want to test the service layer and DAL layer. 
The reason we prefer using an in memory database is because nearly all of the data we are using changes constantly, and this breaks our tests very frequently. So we want to test very specific data scenarios, and setting the in memory database will always run that same test. So if something fails, we know it's a bug in our code vs. data changing in the real database.
Some of our business logic has some very complex checks based of what data we are getting for our business users.
I will work with the example you provided and implement that in our tests.
Thanks again!

**Comment by rgvlee** [comment]
Closing the issue as it appears to be resolved. Let me know if you need further assistance.
